tools/respecDocWriter.js

```diff
@@ -77,6 +77,41 @@ async function fetchAndWrite(
       warn: false,
     };
     handleConsoleMessages(haltFlags);
+    // Intercept requests for ReSpec and serve it from the local build instead.
+    // https://github.com/tidoust/reffy/pull/212
+    await page.setRequestInterception(true);
+    page.on("request", async function callback(request) {
```

Comment:
@sidvishnoi, I believe your netlify work does similar work to load the spec with the specified build, am I right?

Reply:
Yes, but in a service worker.
https://respec-preview.netlify.com/

Reply:
Is that https://github.com/w3c/respec/blob/develop/tools/netlify.js? Where's the logic to replace the ReSpec version?

Are there changes to ReSpec itself that could be made to make this kind of replacing more robust? I'm thinking maybe having the script load but do nothing, or post a message saying "please inject your own ReSpec now"?

Reply:
Using a regex to replace commonly used ReSpec scripts: https://github.com/sidvishnoi/respec-preview/blob/master/sw.js

Reply:
I see, that's not far from what I'm doing, both hostname and the first part of the pathname need to match exactly. My version would also match anything else that's in the build/ directory, but I don't know that it's needed. The code I have now would only replace the first request so I expect it's actually broken in some cases.